### PR TITLE
1742 User can be now updated in AuthenticatedURLSession class

### PR DESCRIPTION
**Pull request by zamorastp**

### CODE REVIEW

File: Sources/AccountSDKIOSWeb/Lib/HTTP/AuthenticatedURLSession.swift [hunk 1]
```diff
@@ -2,7 +2,7 @@ import Foundation
 
 /// AuthenticatedURLSession wraps a User to allow Bearer authenticated requests and the use of URLSessionDataTask
 public final class AuthenticatedURLSession {
-    private let user: User
```
Comment (Daniel-Echegaray) [inline Sources/AccountSDKIOSWeb/Lib/HTTP/AuthenticatedURLSession.swift:5]: Is this a good idea?

Comment (zamorastp) [reply Sources/AccountSDKIOSWeb/Lib/HTTP/AuthenticatedURLSession.swift:5]: I was pretty sure we agreed on that 😅 It was a conclusion from https://sch-chat.slack.com/archives/G011N453QG1/p1635970122017100 

If you see how the user property is used inside the class you can see that it's not going to break anything. There is no special configuration etc. Every time you call dataTask, it just gets tokens from the user and uses them to authenticate request. 
but yes, it is only useful when someone log in second time :) So up to you: we can merge it or forget about it :D

Comment (Daniel-Echegaray) [reply Sources/AccountSDKIOSWeb/Lib/HTTP/AuthenticatedURLSession.swift:5]: The alternative would be that will have to reinstantiate a class of `AuthenticatedURLSession`?

Comment (zamorastp) [reply Sources/AccountSDKIOSWeb/Lib/HTTP/AuthenticatedURLSession.swift:5]: Exactly. Not that bad at all

**Review by Daniel-Echegaray — APPROVED** [review]

**Comment by zamorastp** [comment]
Agreed on not merging it.

**Pull request closed without merging**